include/iotc.h

```diff
@@ -30,224 +30,192 @@ extern "C" {
+ * MQTT with TLS. Embedded systems rely on unique hardware, so the SDK abstracts
+ * these dependencies in the <a href="../../bsp/html/index.html">Board Support
+ * Package</a> (BSP).
+ *
+ * The SDK has a turn-key POSIX BSP, so it builds natively on POSIX platforms.
+ * You can customize the BSP to port the SDK to non-POSIX platforms.
  */
 
 /* -----------------------------------------------------------------------
  * MAIN LIBRARY FUNCTIONS
  * ----------------------------------------------------------------------- */
 
 /**
- * @brief Initialize libraries. Required operation.
- *
- * Applications should call this first when starting a new runtime.
- *
- * @retval IOTC_STATE_OK Status OK.
+ * @details Initializes the BSP time and random number libraries. Applications
+ * must call the function first when starting a new runtime.
  */
 extern iotc_state_t iotc_initialize();
 
 /**
- * @brief Free all resources created during initialization.
- *
- * Call to shut down the Device SDK and to free any of its resources.
- *
- * <b>Note</b>: Clean up contexts individually with
- * <code>iotc_delete_context()</code> before calling this function.
- *
- * @see iotc_initialize
- * @see iotc_create_context
- * @see iotc_delete_context
- * @see iotc_shutdown_connection
- *
- * @retval IOTC_STATE_OK Status OK.
- * @retval IOTC_FAILED_INITIALIZATION An urecoverable error occurred.
+ * @details Shuts down the Device SDK and frees all resources created during
+ * {@link iotc_initialize() initialization}. {@link iotc_delete_context() Free}
+ * all contexts before calling the function.
  */
 extern iotc_state_t iotc_shutdown();
 
 /**
- * @brief Create a connection context.
- *
- * Invoke after initalizing <code>libiotc</code> through
- *      <code>iotc_initialize()</code>.
- *
- * @see iotc_initialize
- * @see iotc_delete_context
- *
- * @retval nonnegative_number a valid context handle is created.
- * @retval negative_number If the function fails, it returns the
- *     <a href="~/include/iotc_error.h">error code</a> multiplied by -1.
+ * @details Creates a connection context. If the function fails, it returns the
+ * opposite of the numeric {@link ::iotc_state_t error code}.
  */
 extern iotc_context_handle_t iotc_create_context();
 
 /**
  * @brief Frees the provided context.
  *
- * Invoke to free memory after the client application disconnects
- * from Cloud IoT Core.
- *
- * You may reuse disconnected contexts to reconnect to Cloud IoT Core;
- * you don't need to destroy and recreate contexts.
+ * @details On POSIX systems, the client application must free the context
+ * after iotc_process_blocking() returns.
  *
- * Do not delete a context in the connection callback.
- *
- * On POSIX systems, the client application must delete the context after
- * <code>iotc_process_blocking()</code> returns.
- *
- * On non-POSIX systems, the application must delete the context on the
+ * On non-POSIX systems, the application must free the context on the
```

Comment:
Here too, this isn't non-POSIX specific, but if the event engine is driven in non-blocking way by the iotc_events_process_tick async then the application must free...

Reply:
Done